Clean up build warnings in DragonSparkFlex and DragonSparkMax

Warnings in file DragonSparkFlex.cpp ....
In file included from C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp\hw\DragonSparkFlex.cpp:17:
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp/hw/DragonSparkFlex.h: In constructor 'DragonSparkFlex::DragonSparkFlex(int, RobotElementNames::MOTOR_CONTROLLER_USAGE, rev::CANSparkLowLevel::MotorType, rev::SparkRelativeEncoder::Type, rev::SparkLimitSwitch::Type, rev::SparkLimitSwitch::Type, const DistanceAngleCalcStruc&)':
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp/hw/DragonSparkFlex.h:103:37: warning: 'DragonSparkFlex::m_feedbackType' will be initialized after [-Wreorder]
  103 |     rev::SparkRelativeEncoder::Type m_feedbackType;
      |                                     ^~~~~~~~~~~~~~
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp/hw/DragonSparkFlex.h:96:33: warning:   'rev::SparkLimitSwitch::Type DragonSparkFlex::m_forwardType' [-Wreorder]
   96 |     rev::SparkLimitSwitch::Type m_forwardType;
      |                                 ^~~~~~~~~~~~~
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp\hw\DragonSparkFlex.cpp:23:1: warning:   when initialized here [-Wreorder]
   23 | DragonSparkFlex::DragonSparkFlex(int id,
      | ^~~~~~~~~~~~~~~
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp/hw/DragonSparkFlex.h:105:29: warning: 'DragonSparkFlex::m_pidController' will be initialized after [-Wreorder]
  105 |     rev::SparkPIDController m_pidController;
      |                             ^~~~~~~~~~~~~~~
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp/hw/DragonSparkFlex.h:94:27: warning:   'rev::SparkLimitSwitch DragonSparkFlex::m_forwardLimitSwitch' [-Wreorder]
   94 |     rev::SparkLimitSwitch m_forwardLimitSwitch;
      |                           ^~~~~~~~~~~~~~~~~~~~
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp\hw\DragonSparkFlex.cpp:23:1: warning:   when initialized here [-Wreorder]
   23 | DragonSparkFlex::DragonSparkFlex(int id,
      | ^~~~~~~~~~~~~~~
Warnings in file DragonSparkMax.cpp ....
In file included from C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp\hw\DragonSparkMax.cpp:18:
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp/hw/DragonSparkMax.h: In constructor 'DragonSparkMax::DragonSparkMax(int, RobotElementNames::MOTOR_CONTROLLER_USAGE, rev::CANSparkLowLevel::MotorType, rev::SparkRelativeEncoder::Type, rev::SparkLimitSwitch::Type, rev::SparkLimitSwitch::Type, const DistanceAngleCalcStruc&)':
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp/hw/DragonSparkMax.h:106:37: warning: 'DragonSparkMax::m_feedbackType' will be initialized after [-Wreorder]
  106 |     rev::SparkRelativeEncoder::Type m_feedbackType;
      |                                     ^~~~~~~~~~~~~~
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp/hw/DragonSparkMax.h:99:33: warning:   'rev::SparkLimitSwitch::Type DragonSparkMax::m_forwardType' [-Wreorder]
   99 |     rev::SparkLimitSwitch::Type m_forwardType;
      |                                 ^~~~~~~~~~~~~
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp\hw\DragonSparkMax.cpp:26:1: warning:   when initialized here [-Wreorder]
   26 | DragonSparkMax::DragonSparkMax(int id,
      | ^~~~~~~~~~~~~~
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp/hw/DragonSparkMax.h:108:29: warning: 'DragonSparkMax::m_pidController' will be initialized after [-Wreorder]
  108 |     rev::SparkPIDController m_pidController;
      |                             ^~~~~~~~~~~~~~~
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp/hw/DragonSparkMax.h:97:27: warning:   'rev::SparkLimitSwitch DragonSparkMax::m_forwardLimitSwitch' [-Wreorder]
   97 |     rev::SparkLimitSwitch m_forwardLimitSwitch;
      |                           ^~~~~~~~~~~~~~~~~~~~
C:\Users\joewi\Documents\GitHub\2024Crescendo\src\main\cpp\hw\DragonSparkMax.cpp:26:1: warning:   when initialized here [-Wreorder]
   26 | DragonSparkMax::DragonSparkMax(int id,
      | ^~~~~~~~~~~~~~

--- a/src/main/cpp/hw/DragonSparkFlex.h
+++ b/src/main/cpp/hw/DragonSparkFlex.h
@@ -91,19 +91,19 @@ private:
     double GetRotationsWithGearNoOffset() const;
     int m_id;
     rev::CANSparkFlex *m_spark;
-    rev::SparkLimitSwitch m_forwardLimitSwitch;
-    rev::SparkLimitSwitch m_reverseLimitSwitch;
-    rev::SparkLimitSwitch::Type m_forwardType;
-    rev::SparkLimitSwitch::Type m_reverseType;
-    CANSparkBase::ControlType m_controlType;
     double m_outputRotationOffset;
-    int m_slot;
-
     RobotElementNames::MOTOR_CONTROLLER_USAGE m_deviceType;
     rev::SparkRelativeEncoder::Type m_feedbackType;
+    rev::SparkLimitSwitch::Type m_forwardType;
+    rev::SparkLimitSwitch::Type m_reverseType;
     rev::SparkRelativeEncoder m_encoder;
     rev::SparkPIDController m_pidController;
+    rev::SparkLimitSwitch m_forwardLimitSwitch;
+    rev::SparkLimitSwitch m_reverseLimitSwitch;
     DistanceAngleCalcStruc m_calcStruc;
+
+    CANSparkBase::ControlType m_controlType;
+    int m_slot;
 
     rev::CANSparkFlex *GetSparkFlex();
 };

--- a/src/main/cpp/hw/DragonSparkMax.h
+++ b/src/main/cpp/hw/DragonSparkMax.h
@@ -94,19 +94,19 @@ private:
     double GetRotationsWithGearNoOffset() const;
     int m_id;
     rev::CANSparkMax *m_spark;
-    rev::SparkLimitSwitch m_forwardLimitSwitch;
-    rev::SparkLimitSwitch m_reverseLimitSwitch;
-    rev::SparkLimitSwitch::Type m_forwardType;
-    rev::SparkLimitSwitch::Type m_reverseType; // DRAGON_CONTROL_MODE m_controlMode;
-    CANSparkBase::ControlType m_controlType;
     double m_outputRotationOffset;
-    int m_slot;
-
     RobotElementNames::MOTOR_CONTROLLER_USAGE m_deviceType;
     rev::SparkRelativeEncoder::Type m_feedbackType;
+    rev::SparkLimitSwitch::Type m_forwardType;
+    rev::SparkLimitSwitch::Type m_reverseType;
     rev::SparkRelativeEncoder m_encoder;
     rev::SparkPIDController m_pidController;
+    rev::SparkLimitSwitch m_forwardLimitSwitch;
+    rev::SparkLimitSwitch m_reverseLimitSwitch;
     DistanceAngleCalcStruc m_calcStruc;
-    rev::CANSparkMax *
-    GetSparkMax();
+
+    CANSparkBase::ControlType m_controlType;
+    int m_slot;
+
+    rev::CANSparkMax *GetSparkMax();
 };

